tools/pipelines/templates/include-publish-npm-package-deployment.yml

```diff
@@ -42,10 +42,47 @@ parameters:
   default: $(Pipeline.Workspace)/.pnpm-store
 
 jobs:
+
+  # This job is necessary to be compliant with 1ES.
+  # 1ES does not allow checking out a code repo within a deployment job so instead we use this pattern
+  # of checking out the repo in separate job and uploading it as a pipeline artifact.
+- job: checkoutAndUploadThisRepoAsArtifact
+  steps:
+  # Checkout the repository
+  - checkout: self
+    persistCredentials: true # Necessary for creation of git tags to work
```

Comment:
Does this job actually tag the repo?

Reply:
I think tagging remains in the job that does it today (the deployment job that "becomes" a 1ES release job), this extra one wouldn't need this setting. But in https://github.com/microsoft/FluidFramework/pull/23913/files#r1968504949 I argue the new job probably isn't the right solution.